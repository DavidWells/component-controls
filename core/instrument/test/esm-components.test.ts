import { loadStoriesTests } from './loadTestFiles';

describe('esm-components', () => {
  loadStoriesTests(
    {
      stories: { sourceFiles: false, package: false },
      components: { sourceFiles: false, package: false },
    },
    ['esm', 'components'],
  );
});
