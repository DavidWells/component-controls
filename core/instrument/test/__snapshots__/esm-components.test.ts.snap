// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`esm-components default-import.js 1`] = `
Object {
  "components": Object {
    "68aebf519ff8f8359ff912b07a1678ac": Object {
      "from": "../../components/button-default-arrow-func",
      "importedName": "default",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "Button",
      "request": "/Users/atanasster/cc/core/instrument/test/fixtures/components/button-default-arrow-func.js",
    },
  },
  "doc": Object {
    "component": "Button",
    "components": Object {},
    "componentsLookup": Object {
      "Button": "68aebf519ff8f8359ff912b07a1678ac",
    },
    "date": 2020-10-28T15:50:29.497Z,
    "dateModified": 2020-10-28T15:50:29.497Z,
    "title": "Story",
  },
  "packages": Object {},
  "stories": Object {},
  "transformed": "import Button from '../../components/button-default-arrow-func';
export default {
  title: 'Story',
  component: Button,
};
",
}
`;

exports[`esm-components story-subcomponents.js 1`] = `
Object {
  "components": Object {
    "3058c83bfd1ea05ca4b611ea98c7f835": Object {
      "from": "../../components/button-default-arrow-func",
      "importedName": "default",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "ArrowButton",
      "request": "/Users/atanasster/cc/core/instrument/test/fixtures/components/button-default-arrow-func.js",
    },
    "37426d3b0c28836644ead936e512dfa6": Object {
      "from": "../../components/button-named-arrow-func",
      "importedName": "Button",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "Button",
      "request": "/Users/atanasster/cc/core/instrument/test/fixtures/components/button-named-arrow-func.js",
    },
  },
  "doc": Object {
    "components": Object {},
    "componentsLookup": Object {
      "ArrowButton": "3058c83bfd1ea05ca4b611ea98c7f835",
      "Button": "37426d3b0c28836644ead936e512dfa6",
    },
    "date": 2020-10-28T15:50:29.498Z,
    "dateModified": 2020-10-28T15:50:29.498Z,
    "title": "Story",
  },
  "packages": Object {},
  "stories": Object {
    "story": Object {
      "arguments": Array [],
      "component": "ArrowButton",
      "id": "story",
      "loc": Object {
        "end": Object {
          "column": 34,
          "line": 8,
        },
        "start": Object {
          "column": 21,
          "line": 8,
        },
      },
      "name": "story",
      "source": "() => 'hello'",
      "subcomponents": Object {
        "My Button Tab": "Button",
      },
    },
  },
  "transformed": "import ArrowButton from '../../components/button-default-arrow-func';
import { Button } from '../../components/button-named-arrow-func';

export default {
  title: 'Story',
};

export const story = () => 'hello';

story.story = {
  component: ArrowButton,
  subcomponents: { 'My Button Tab': Button },
};
",
}
`;

exports[`esm-components subcomponents.js 1`] = `
Object {
  "components": Object {
    "3058c83bfd1ea05ca4b611ea98c7f835": Object {
      "from": "../../components/button-default-arrow-func",
      "importedName": "default",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "ArrowButton",
      "request": "/Users/atanasster/cc/core/instrument/test/fixtures/components/button-default-arrow-func.js",
    },
    "37426d3b0c28836644ead936e512dfa6": Object {
      "from": "../../components/button-named-arrow-func",
      "importedName": "Button",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "name": "Button",
      "request": "/Users/atanasster/cc/core/instrument/test/fixtures/components/button-named-arrow-func.js",
    },
  },
  "doc": Object {
    "component": "ArrowButton",
    "components": Object {},
    "componentsLookup": Object {
      "ArrowButton": "3058c83bfd1ea05ca4b611ea98c7f835",
      "Button": "37426d3b0c28836644ead936e512dfa6",
    },
    "date": 2020-10-28T15:50:29.498Z,
    "dateModified": 2020-10-28T15:50:29.498Z,
    "subcomponents": Object {
      "Button": "Button",
    },
    "title": "Story",
  },
  "packages": Object {},
  "stories": Object {},
  "transformed": "import ArrowButton from '../../components/button-default-arrow-func';
import { Button } from '../../components/button-named-arrow-func';

export default {
  title: 'Story',
  component: ArrowButton,
  subcomponents: { Button },
};
",
}
`;
